gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/rules/AnyOneOfExecutionEngineRule.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.pivotal.gemfirexd.internal.impl.sql.rules;
+
+import com.pivotal.gemfirexd.internal.engine.GfxdConstants;
+import com.pivotal.gemfirexd.internal.engine.distributed.metadata.DMLQueryInfo;
+import com.pivotal.gemfirexd.internal.engine.distributed.metadata.QueryInfo;
+import com.pivotal.gemfirexd.internal.engine.sql.execute.SnappyActivation;
+import com.pivotal.gemfirexd.internal.iapi.services.sanity.SanityManager;
+
+class AnyOneOfExecutionEngineRule extends ExecutionEngineRule {
+  private enum RuleName {
+    DISTINCT_QUERY_RULE,
+    GROUPBY_QUERY_RULE,
+    UNION_OR_INTERSECT_QUERY_RULE,
+    None
+  }
+
+  RuleName rule = RuleName.None;
+
+  @Override
+  protected boolean findExecutionEngine(DMLQueryInfo qInfo) {
```

Comment:
Find returning a boolean. May be a better name which indicates it is returning some rule flag. Here if the first rule matches still it will come down and do two if condition matching and then again a ref check

rule == RuleName.None

and then  return true or false.

You can return the rule itself andcheck for non nullness of it. That way you don't need to create a rule object to store the rule. Probably its enough to pass around in args.